### PR TITLE
JIRA: VZ-6265 Enable MySQL backed Grafana deployment

### PR DESCRIPTION
**Pull request by jmaron99**
- Enabled specifying an external DB instance for Grafana dashboard persistence (HA)
- Added modified unit/tests
- Added an end to end test and triggered said test from periodic runs

### CODE REVIEW

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go [hunk 1]
```diff
@@ -434,10 +434,18 @@ type KubeStateMetricsComponent struct {
...
+type DatabaseInfo struct {
+	Host string `json:"host,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
 // GrafanaComponent specifies the Grafana configuration.
 type GrafanaComponent struct {
 	// +optional
-	Enabled *bool `json:"enabled,omitempty"`
+	Enabled  *bool         `json:"enabled,omitempty"`
```
Comment (stevshap) [inline platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:446]: With these changes to the api, do install profiles need to be updated with enabled: true/false values?

File: platform-operator/controllers/verrazzano/component/grafana/vmi.go [hunk 2]
```diff
@@ -31,12 +33,26 @@ func newGrafana(cr *vzapi.Verrazzano, storage *common.ResourceRequestValues, exi
...
 	}
+	if grafanaSpec.Database != nil {
+		log.Infof("Configuring database info: %v", grafanaSpec.Database)
+		grafana.Database = &vmov1.Database{
+			PasswordSecret: constants.GrafanaDBSecret,
+			Host:           grafanaSpec.Database.Host,
+			Name:           grafanaSpec.Database.Name,
+		}
+	}
+	log.Infof("VMO grafana spec: %v", grafana)
```
Comment (stevshap) [inline platform-operator/controllers/verrazzano/component/grafana/vmi.go:56]: Should this only be logged in the Database != nil case?

File: tests/testdata/grafana/grafana-mysql.yaml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            periodSeconds: 10
+            timeoutSeconds: 5
+            successThreshold: 1
+            failureThreshold: 3
+          readinessProbe:
+            exec:
+              command:
+                - sh
+                - -c
+                - "mysqladmin ping -u root -p${MYSQL_ROOT_PASSWORD}"
```
Comment (stevshap) [inline tests/testdata/grafana/grafana-mysql.yaml:125]: Does this work? Should there be a space between -p and ${MYSQL...}

File: platform-operator/controllers/verrazzano/component/grafana/vmi.go (hunk 2 above)
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/component/grafana/vmi.go:56]: It's more of an attempt to generally assess the state of the VMO spec for grafana.  Can probably be changed to debugf.

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go (hunk 1 above)
Comment (jmaron99) [reply platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:446]: Not sure what you mean.  the value defaults to true much like other components.

File: platform-operator/controllers/verrazzano/component/grafana/vmi.go (hunk 2 above)
Comment (stevshap) [reply platform-operator/controllers/verrazzano/component/grafana/vmi.go:56]: I think debug is better 👍🏼

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go (hunk 1 above)
Comment (stevshap) [reply platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:446]: Ok that's all I was wondering

File: go.sum [hunk 4]
```diff
@@ -956,6 +956,16 @@ github.com/urfave/cli v1.22.1/go.mod h1:Gos4lmkARVdJ6EkW0WaNv/tZAAMe9V7XWyB60NtX
 github.com/urfave/cli v1.22.2/go.mod h1:Gos4lmkARVdJ6EkW0WaNv/tZAAMe9V7XWyB60NtXRu0=
 github.com/verrazzano/verrazzano-monitoring-operator v0.0.29-0.20220706002248-a07bafbdc991 h1:hWwKJXrSkmUbsfCl4wjJs+yWmWwFb/iQsM155syAIk4=
 github.com/verrazzano/verrazzano-monitoring-operator v0.0.29-0.20220706002248-a07bafbdc991/go.mod h1:Tpaz1662UlhEvo9A5nUKCi+NcCrrZ1elkOBukyerktc=
+github.com/verrazzano/verrazzano-monitoring-operator v0.0.29-0.20220715142548-a838f721d905 h1:dyFeN4SKuhtDN6EqzAtzm2s+DjmvF6V9J/NoiAm2Jv8=
```
Comment (mgianatagh) [inline go.sum:959]: it seems odd that so many entries were added for VMO

File: tests/e2e/config/scripts/process_grafana_db_install_yaml.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# Copyright (c) 2020, 2022, Oracle and/or its affiliates.
```
Comment (mgianatagh) [inline tests/e2e/config/scripts/process_grafana_db_install_yaml.sh:3]: If this is a new file, the copyright year should just be "2022,"

**Review by mgianatagh — DISMISSED** [review]

File: go.sum (hunk 4 above)
Comment (jmaron99) [reply go.sum:959]: Agreed.  but not something I seem to be able to control.

**Review by mgianatagh — APPROVED** [review]

**Review by stevshap — APPROVED** [review]

**Pull request merged**